### ISSUE TITLE
running silverpeas error

### ISSUE THREAD

**Issue by chenliu1993** [Silverpeas/docker-silverpeas-prod#3]
Hi devs, 
I am using silverpeas acording to the dockerhub page. After postgres is startting, I then docker run silverpeas, but I run into some errors like below:
```
2020-03-23 19:50:39,988 ERROR [org.jboss.as.controller.management-operation] (Controller Boot Thread) WFLYCTL0190: Step handler org.jboss.as.server.DeployerChainAddHandler$FinalRuntimeStepHandler@498c42f2 for operation {"operation" => "add-deployer-chains","address" => []} at address [] failed handling operation rollback -- java.util.concurrent.TimeoutException: java.util.concurrent.TimeoutException
	at org.jboss.as.controller.OperationContextImpl.waitForRemovals(OperationContextImpl.java:498)
	at org.jboss.as.controller.AbstractOperationContext$Step.handleResult(AbstractOperationContext.java:1369)
	at org.jboss.as.controller.AbstractOperationContext$Step.finalizeInternal(AbstractOperationContext.java:1328)
	at org.jboss.as.controller.AbstractOperationContext$Step.finalizeStep(AbstractOperationContext.java:1301)
	at org.jboss.as.controller.AbstractOperationContext$Step.access$300(AbstractOperationContext.java:1185)
	at org.jboss.as.controller.AbstractOperationContext.executeResultHandlerPhase(AbstractOperationContext.java:767)
	at org.jboss.as.controller.AbstractOperationContext.processStages(AbstractOperationContext.java:644)
	at org.jboss.as.controller.AbstractOperationContext.executeOperation(AbstractOperationContext.java:370)
	at org.jboss.as.controller.OperationContextImpl.executeOperation(OperationContextImpl.java:1329)
	at org.jboss.as.controller.ModelControllerImpl.boot(ModelControllerImpl.java:493)
	at org.jboss.as.controller.AbstractControllerService.boot(AbstractControllerService.java:387)
	at org.jboss.as.controller.AbstractControllerService.boot(AbstractControllerService.java:349)
	at org.jboss.as.server.ServerService.boot(ServerService.java:397)
	at org.jboss.as.server.ServerService.boot(ServerService.java:366)
	at org.jboss.as.controller.AbstractControllerService$1.run(AbstractControllerService.java:299)
	at java.lang.Thread.run(Thread.java:748)
```
and before thie line is printed, a `300 Timeout` issue was found.
More upper in the log , I found some process in container are trying to connect to somthing
```
See tika-parsers/pom.xml for the correct version.
2020-03-23 19:46:21,751 INFO  [org.apache.jackrabbit.core.query.lucene.MultiIndex] (MSC service thread 1-1) indexing... /jcr:system/jcr:nodeTypes/rep:Token/jcr:propertyDefinition (100)
2020-03-23 19:46:22,349 INFO  [org.apache.jackrabbit.core.query.lucene.SearchIndex] (MSC service thread 1-1) Index initialized: /opt/silverpeas/data/jcr/repository/index Version: 3
2020-03-23 19:46:26,256 INFO  [org.apache.jackrabbit.core.query.lucene.SearchIndex] (MSC service thread 1-1) Index initialized: /opt/silverpeas/data/jcr/workspaces/silverpeas/index Version: 3
2020-03-23 19:46:26,257 INFO  [org.apache.jackrabbit.core.RepositoryImpl] (MSC service thread 1-1) workspace 'silverpeas' initialized
2020-03-23 19:46:26,265 INFO  [org.apache.jackrabbit.core.RepositoryImpl] (MSC service thread 1-1) SecurityManager = class org.apache.jackrabbit.core.security.simple.SimpleSecurityManager
2020-03-23 19:46:26,269 INFO  [org.apache.jackrabbit.core.security.simple.SimpleSecurityManager] (MSC service thread 1-1) init: using Repository LoginModule configuration for Silverpeas
2020-03-23 19:46:26,282 INFO  [org.apache.jackrabbit.core.RepositoryImpl] (MSC service thread 1-1) Repository started (12352ms)
2020-03-23 19:46:26,284 INFO  [org.jboss.as.connector.deployers.RaXmlDeployer] (MSC service thread 1-1) jackrabbit-jca.rar: Created repository (org.apache.jackrabbit.core.RepositoryImpl@115b9bb2)
2020-03-23 19:46:26,288 INFO  [org.jboss.as.connector.deployers.RaXmlDeployer] (MSC service thread 1-1) jackrabbit-jca.rar: Created repository handle (org.apache.jackrabbit.jca.JCARepositoryHandle@6a6b8f72)
2020-03-23 19:46:26,290 INFO  [org.jboss.as.connector.deployment] (MSC service thread 1-1) WFLYJCA0007: Registered connection factory java:/jca/jcr
2020-03-23 19:46:26,296 WARN  [org.jboss.as.connector.deployers.RaXmlDeployer] (MSC service thread 1-1) IJ020016: Missing <recovery> element. XA recovery disabled for: java:/jca/jcr
2020-03-23 19:46:26,300 INFO  [org.jboss.as.connector.deployers.RaXmlDeployer] (MSC service thread 1-1) IJ020002: Deployed: file:/opt/wildfly/standalone/tmp/vfs/temp/temp3e4f4afea4f1a29b/jackrabbit-jca.rar-4b9233a93549af16/contents/
2020-03-23 19:46:26,305 INFO  [org.jboss.as.connector.deployment] (MSC service thread 1-4) WFLYJCA0002: Bound JCA ConnectionFactory [java:/jca/jcr]

2020-03-23 19:48:15,833 WARN  [org.jboss.as.server.deployment] (MSC service thread 1-2) WFLYSRV0059: Class Path entry ridl.jar in /content/silverpeas.war/WEB-INF/lib/juh-3.0.1.jar  does not point to a valid jar for a Class-Path reference.
2020-03-23 19:48:15,840 WARN  [org.jboss.as.server.deployment] (MSC service thread 1-2) WFLYSRV0059: Class Path entry jurt.jar in /content/silverpeas.war/WEB-INF/lib/juh-3.0.1.jar  does not point to a valid jar for a Class-Path reference.
2020-03-23 19:48:15,846 WARN  [org.jboss.as.server.deployment] (MSC service thread 1-2) WFLYSRV0059: Class Path entry ../../lib/ in /content/silverpeas.war/WEB-INF/lib/juh-3.0.1.jar  does not point to a valid jar for a Class-Path reference.
2020-03-23 19:48:15,849 WARN  [org.jboss.as.server.deployment] (MSC service thread 1-2) WFLYSRV0059: Class Path entry ../bin/ in /content/silverpeas.war/WEB-INF/lib/juh-3.0.1.jar  does not point to a valid jar for a Class-Path reference.
2020-03-23 19:48:15,854 WARN  [org.jboss.as.server.deployment] (MSC service thread 1-2) WFLYSRV0059: Class Path entry ridl.jar in /content/silverpeas.war/WEB-INF/lib/jurt-3.0.1.jar  does not point to a valid jar for a Class-Path reference.
2020-03-23 19:48:15,858 WARN  [org.jboss.as.server.deployment] (MSC service thread 1-2) WFLYSRV0059: Class Path entry unoloader.jar in /content/silverpeas.war/WEB-INF/lib/jurt-3.0.1.jar  does not point to a valid jar for a Class-Path reference.
2020-03-23 19:48:15,860 WARN  [org.jboss.as.server.deployment] (MSC service thread 1-2) WFLYSRV0059: Class Path entry ../../lib/ in /content/silverpeas.war/WEB-INF/lib/jurt-3.0.1.jar  does not point to a valid jar for a Class-Path reference.
2020-03-23 19:48:15,865 WARN  [org.jboss.as.server.deployment] (MSC service thread 1-2) WFLYSRV0059: Class Path entry ../bin/ in /content/silverpeas.war/WEB-INF/lib/jurt-3.0.1.jar  does not point to a valid jar for a Class-Path reference.
```
But i found there is no /content files under the /.
What problems could this be? Is this because I am using it in a wrong way?
Thanks.

**Comment by chenliu1993** [comment]
I am running in a 2 core 4 GB VM, and postgres is running in another vm in host mode.

**Comment by chenliu1993** [comment]
And I am connecting to the postgres with a -e DB_SERVER=ip, not an --link, could this be a problem?

**Comment by mmoqui** [comment]
Hi Lui Chen,

I've just checked your way to start a container with Silverpeas and I didn't encounter any problems. The machine in which the Docker container with Silverpeas is ran is a VM configured as such:

- 4 CPU
- 4 Go RAM

Below the steps I did:

1. In a host in which PostgreSQL is installed I created a database silverpeas and a user silverpeas, and I granted to this user all privileges on the database silverpeas;
2. I updated the /etc/postgresql/10/main/pg_hba.conf configuration file by adding an entry authorizing the remote host on which Silverpeas will be ran to access the database. PostgreSQL is then restarted;
4. I updated the firewall rules in the two machines so that Silverpeas can access the database;
3. In the host with Docker and on which Silverpeas has to be ran, I executed this command line:
```bash
$docker run --name silverpeas -p 8080:8000 -d  \
-e DB_NAME="silverpeas" \
-e DB_USER="silverpeas" \
-e DB_PASSWORD="<the password of the db user silverpeas>" 
-e DB_SERVER="<the IP address of the host running PostgreSQL>" \
-v silverpeas-log:/opt/silverpeas/log \
-v silverpeas-data:/opt/silverpeas/data \
silverpeas
```

And all are working fine.

Now my guest about your timeout problem. When a container is spawn from a Silverpeas Docker image, a post-installation is performed during which Wildfly and the database will be setup according to the execution environment. Then, once done, Silverpeas is started. In some circumstances (mainly in machines with low resources), we encountered a problem either during the post-installation or at Silverpeas starting: Wildfly takes too long to start and after some times the installation/starting program raises a timeout and stops. This issues is fixed in the future Silverpeas 6.1.For instance, no Docker image of a release-candidate of this version is yet generated.
